.github/workflows/maven.yml

```diff
@@ -18,12 +18,12 @@ jobs:
 
     steps:
       - name: Checkout sources
-        uses: actions/checkout@v2.3.4
+        uses: actions/checkout@v4
```

Comment:
you need to put the hash corresponding to the release, as the tag might be changed by an attacker

Reply:
You can look at powsybl/powsybl-core#2821 for an example

Reply:
Ok, thanks for pointing it out. I took more recent version too, following the example in core: 
- 4.1.1 for actions/checkout instead of 4
- 4.0.0 for actions/setup-java instead of 3